overlord/devicestate/devicemgr.go

```diff
@@ -849,7 +849,21 @@ func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAct
 		return ErrUnsupportedAction
 	}
 
-	// TODO:UC20 update boot environment and schedule a reboot
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	deviceCtx, err := DeviceCtx(m.state, nil, nil)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+
+	if err := boot.SetRecoveryBootSystemAndMode(deviceCtx, systemLabel, action.Mode); err != nil {
+		return fmt.Errorf("cannot boot into system %q in mode %q: %v",
+			systemLabel, action.Mode, err)
+	}
+
+	logger.Noticef("restarting into system %q for action %q", systemLabel, sysAction.Title)
+	m.state.RequestRestart(state.RestartSystem)
```

Comment:
this should use RestartSystemNow right, when available ?

Reply:
Yes, I hoped to land #8386 quickly, but it seems that PR must reach a decent number of CI restarts before this can happen.

Reply:
The PR has landed, I've merged master and pushed the patches that request the new immediate reboot.